pkg/chain/ethereum/tbtc.go

```diff
@@ -8,52 +8,138 @@ import (
 	"sort"
 
 	chain "github.com/keep-network/keep-ecdsa/pkg/chain"
+	"github.com/keep-network/keep-ecdsa/pkg/chain/gen/ethereum/contract"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/keep-network/keep-common/pkg/chain/ethereum/ethutil"
 	"github.com/keep-network/keep-common/pkg/subscription"
-	"github.com/keep-network/tbtc/pkg/chain/ethereum/gen/contract"
+	tbtcchain "github.com/keep-network/tbtc/pkg/chain/ethereum/gen/contract"
 )
 
-// TBTCEthereumChain represents an Ethereum chain handle with
+// tbtcApplication represents an Ethereum chain handle with
 // TBTC-specific capabilities.
-type TBTCEthereumChain struct {
-	*ethereumChain
+type tbtcApplication struct {
+	chainHandle *ethereumChain
 
-	tbtcSystemContract *contract.TBTCSystem
+	bondedECDSAKeepFactoryContract *contract.BondedECDSAKeepFactory
```

Comment:
`bondedECDSAKeepFactoryContract` is not set during `tbtcApplication` initialization in `TBTCApplicationHandle` method. That causes some panics in the working app.

Reply:
This is fixed.